### ISSUE TITLE
Web frontend for GitArena

### ISSUE THREAD

**Issue by mellowagain** [mellowagain/gitarena#17]
GitArena is currently lacking major parts of it's web frontend. The existing sites are also currently not very pretty and need cleanup. **Help is greatly appreciated!**

The following sites still need to be implemented:

- [ ] Home page (`/`, logged out view)
- [ ] User dashboard (`/`, logged in view)
- [ ] Explore page (`/explore`): 
    - [x] Standard page to view public & internal repositories as well as a few simple filters
    - [ ] Current activity (trending)
    - [ ] "Featured repositories" (configurable via admin panel)
    - Sort of like [GitHub](https://github.com/explore) and [GitLab](https://gitlab.com/explore)
- [x] Repository pages (`/{username}/{repo_name}`)
    - [ ] Issues list (`/.../issues`)
        - [ ] Specific issue view (`/.../issues/{nr}`) 
    - [ ] Merge requests list (`/.../mergers`)
        - [ ] Specific merge request view (`.../mergers/{nr}`) 
    - [ ] Releases list (`/.../releases`, based on Git tags)
        - [ ] Specific release view (`/.../releases/{tag_name}`) 
    - [x] Commit list (`.../tree/{branch}/commits`)
        - [ ] Specific commit view (`/.../commit/{oid}`
    - [x] Specific directory view (`/.../tree/{branch}/directory/{dir_name}`
    - [x] Specific file view (`/.../tree/{branch}/blob/{file_name}`
- [ ] Users view (`/{username}`)
- [ ] ... 

And many more! List will be updated once progress on existing pages has increased.

**Comment by mellowagain** [comment]
There has been a question regarding frontend support for environments without activated JavaScript. Here are my thoughts regarding this issue:

There is some thought put into making GitArena working without JavaScript, for example:

* Infinite scroll pages (such as commit listing) contain a `<noscript>` fallback which includes buttons for manual navigation.
* XHR links such as starring a repository contain a `<noscript>` fallback with a `<button>` form submit. 

The problem is that some features are simply too much effort to implement without JS or are just straight up impossible to implement without JS, for example:

* Copy to clipboard buttons

There are also some places where support for non-JavaScripts clients was deliberately ignored, for example:

* README rendering in the repository view and directory views

Regardless, I am a believer that at least the basics should function without JavaScript. Apart from belief or stubbornness, there are some very valid reasons one has JavaScript disabled such as for example company policies or a very slow connection. The web has largely ignored these people and in my opinion that is wrong. But a small minority of people should also not remove conviences from the majority of people.

TLDR: **The basics should work without JavaScript. Everything else is a nice to have but only as long as it does not inconvient the majority of users which do in fact have JavaScript enabled.**

In case you are one of the users that do in fact have JavaScript disabled and you encounter an issue with GitArena, feel free to open an issue in this repository. Please do keep in mind that it may be prioritized as low priority as this environment is very niche.

**Comment by mellowagain** [comment]
this issue described the old web frontend from which we've moved away
